notify/impl.go

```diff
@@ -474,17 +488,42 @@ func (n *PagerDuty) Notify(ctx context.Context, as ...*types.Alert) (bool, error
 		details[k] = tmpl(v)
 	}
 
-	msg := &pagerDutyMessage{
-		ServiceKey:  tmpl(string(n.conf.ServiceKey)),
-		EventType:   eventType,
-		IncidentKey: hashKey(key),
-		Description: tmpl(n.conf.Description),
-		Details:     details,
+	msg := &pagerDutyMessage{}
+
+	if n.conf.ServiceKey != "" {
```

Comment:
Could you branch here into two separate functions, something like 
```go
if n.conf.ServiceKey != "" {
	level.Info(n.logger).Log("msg", "PagerDuty v1 API will no longer be supported: https://v2.developer.pagerduty.com/v2/docs/api-v2-frequently-asked-questions")
	return n.notifyV1(ctx, as...)
else {
        return n.notifyV2(ctx, as...)
}
```

And have the same for the retry logic?